### ISSUE TITLE
Update implementation to reflect latest NIST PubID scheme

### ISSUE THREAD

**Issue by ronaldtse** [metanorma/pubid-nist#153]
The NIST Information Services Office has published the final version of the PubID scheme:
https://www.nist.gov/system/files/documents/2022/04/01/PubID_Syntax_NIST_TechPubs.pdf

We need to check the changes and tick off issues that relate to syntax in this repository.

**Comment by mico** [comment]
@ronaldtse There are a lot of series we use that are not in "A.1 Series Abbreviations" list, just some of missing ones:

```
  NBS BRPD-CRPL-D: Basic Radio Propagation Predictions Series
  NBS CRPL: Central Radio Propagation Laboratory Reports
  NBS CRPL-F-A: CRPL Ionospheric Data
  NBS CRPL-F-B: CRPL Solar-Geophysical Data
  NBS IP: CRPL Ionospheric Predictions
  NBS CIS: Consumer Information Series
  NBS CS-E: Commercial Standards (emergency)
  NBS IRPL: Interservice Radio Propagation Laboratory
  ITL Bulletin: ITL Bulletin
  NIST LC: Letter Circular
  NBS LC: Letter Circular
```

Also, there are no distinction between NBS or NIST publisher, does it mean we should support the same series for both of them?

There are also "A.2 Special Publications Subseries".

We have missing "SP 955" and "SP 1011" – that's what I briefly found. Is that an issue?

**Comment by ronaldtse** [comment]
> There are a lot of series we use that are not in "A.1 Series Abbreviations" list, just some of missing ones:

I will let NIST know.

> Also, there are no distinction between NBS or NIST publisher, does it mean we should support the same series for both of them?

No, we should keep our current method.

> There are also "A.2 Special Publications Subseries".

We should probably create an enumeration of these entries. Do we parse all of these? If we do then it's okay.

> We have missing "SP 955" and "SP 1011" – that's what I briefly found. Is that an issue?

What does this mean?

**Comment by mico** [comment]
> > There are also "A.2 Special Publications Subseries".
> 
> We should probably create an enumeration of these entries. Do we parse all of these? If we do then it's okay.
For `NIST SP 800-90r1` we parse "800" as a document number, not as series and series as "NIST SP"

> > We have missing "SP 955" and "SP 1011" – that's what I briefly found. Is that an issue?
> 
> What does this mean?
We don't have "SP 955" and "SP 1011" series in "Special Publications Subseries" list. But probably more than just these two.

**Comment by mico** [comment]
`NBS CRPL 4-m-5` (sequence number (part of report number) cannot have "m", what else "m" can be?)
`NBS IR 80-2073.3` – part number or sequence number cannot have "."

**Comment by mico** [comment]
Now we convert identifiers with year and month to edition, like this:
`NBS FIPS 14-1-Dec1980` -> `NBS FIPS 14-1e198012`
`NBS FIPS 107-Mar1985` -> `NBS FIPS 107e198503`

But PubID 1.0 edition don't allow to use month:
```
<edition> = <edition-type><edition-id>
<edition-type> = ″-″ | ″e″ | ″r″
<edition-id> = {1-9} | yyyy
```

**Comment by mico** [comment]
PubID 1.0 not covering NBS RPT identifiers:
```
NBS RPT Oct-Dec1950
NBS RPT ADHOC
```

Other identifiers not covered by PubID 1.0:
```
NBS SP 535v2a-l
NBS CS 102E-42 (102-42E can work if "E" is a part)
NBS RPT 4817-A (if "A" is a part "4817A" can work)
```

**Comment by mico** [comment]
`part-id` cannot have "-": `NBS CRPL 1-2pt3-1` (original `NBS CRPL 1-2_3-1`)

`NBS CSM v6pt1` – I'm not sure if PubID allow having identifiers without report number

`NBS CIRC 13e2rJune1908` (original `NBS CIRC 13e2revJune1908`) – edition cannot have month

**Comment by mico** [comment]
> @mico anything else we need to do here? For the data source issues we have to file them elsewhere.

@ronaldtse Nothing else we can do, only need to clarify what to do with identifiers that not fits for PubID 1.0

**Comment by ronaldtse** [comment]
I see. Have we implemented PubID 1.0 now?

**Comment by ronaldtse** [comment]
If only the special identifiers are left, let's create a new ticket for "Identifiers that don't fit PubID 1.0" and create individual tickets inside to track each data source issue.

**Comment by mico** [comment]
> I see. Have we implemented PubID 1.0 now?

Yes, but there are some exceptions mentioned above, because not all current identifiers fits for PubID 1.0

**Comment by ronaldtse** [comment]
@mico if we have already fully implemented PubID 1.0, can you please close this? Thanks.
